### PR TITLE
[@types/auth0] Include the missing methods/entities for auth0 enrollment status

### PR DESCRIPTION
**Pull request by cviquez-sama**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/auth0/index.d.ts [hunk 1]
```diff
@@ -1277,6 +1289,17 @@ export class ManagementClient<A=AppMetadata, U=UserMetadata> {
...
   deleteCustomDomain(params: ObjectWithId): Promise<void>;
   deleteCustomDomain(params: ObjectWithId, cb: (err: Error) => void): void;
+
+  // User enrollment
+  getGuardianEnrollments(): Promise<Enrollment[]>;
+  getGuardianEnrollments(cb: (err: Error, response: Enrollment[]) => void): void;
+
+
+  getGuardianEnrollment(params: ObjectWithId): Promise<Enrollment[]>;
+  getGuardianEnrollment(params: ObjectWithId, cb: (err: Error, response: Enrollment[]) => void): void;
```
Comment (peterblazejewicz) [inline types/auth0/index.d.ts:1299]: are those supposed to return a list? or single object?
https://github.com/auth0/node-auth0/blob/c509610693ccec51fb17c586eafe57c6e708e7d2/src/management/GuardianManager.js#L158

Comment (cviquez-sama) [reply types/auth0/index.d.ts]: It should return an Enrollment list. I already update it and now it is working clear with my project.
Also I realized that I included a unnecessary method getGuardianEnrollment and getGuardianEnrollments (in plural), I update this and the singular method is no longer needed

By the way Thanks for the review, 
I will be aware of your reply

File: types/auth0/auth0-tests.ts [hunk 2]
```diff
@@ -633,3 +633,10 @@ management.verifyCustomDomain({ id: 'cd_0000000000000001' }, (err, domainVerific
 
 management.deleteCustomDomain({ id: 'cd_0000000000000001' }).then(() => console.log('deleted'));
 management.deleteCustomDomain({ id: 'cd_0000000000000001' }, err => console.log('deleted'));
+
+// User enrollment
+management.getGuardianEnrollments({ id: 'cd_0000000000000001' }).then((enrollment) => console.log(enrollment))
+management.getGuardianEnrollments({ id: 'cd_0000000000000001' },(err, enrollment) => console.log(enrollment))
```
Comment (peterblazejewicz) [inline types/auth0/auth0-tests.ts:639]:
```suggestion
management.getGuardianEnrollments({ id: 'cd_0000000000000001' }).then((enrollments) => console.log(enrollments))
management.getGuardianEnrollments({ id: 'cd_0000000000000001' },(err, enrollments) => console.log(enrollments))
```
so this a list, shoudl ends with `s`, to point what's being returned. Am I correct?

Comment (cviquez-sama) [reply types/auth0/auth0-tests.ts]: Updated

Yes, it should be in plural to make it clearer

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@cviquez-sama thx!

**Comment by cviquez-sama** [comment]
Ready to merge

**Pull request merged**